src/mutate-errors-to-warnings.ts

```diff
@@ -0,0 +1,32 @@
+import { generateFileName, _buildTodoDatum } from '@ember-template-lint/todo-utils';
+import { TodoData } from '@ember-template-lint/todo-utils/lib/types';
+import { ESLint } from 'eslint';
+
+/**
+ * Mutates all errors present in the todo map to warnings in the results array.
+ * If todo map is not passed, all errors are converted to warnings.
+ * @param results ESLint results array
+ * @param todoMap Optional map of a todoDatum hash and its data.
+ */
+export function mutateTodoErrorsToWarnings(
+  results: ESLint.LintResult[],
+  todoMap?: Map<string, TodoData>
+): void {
+  results.forEach((result: ESLint.LintResult) => {
+    result.messages.forEach((message) => {
+      if (message.severity === 2) {
+        const todoDatum = _buildTodoDatum(result, message);
+        const todoHash = generateFileName(todoDatum);
+        if (!todoMap || todoMap.get(todoHash)) {
+          message.severity = 1;
+          result.errorCount -= 1;
+          result.warningCount += 1;
+          if (message.fix) {
+            result.fixableErrorCount -= 1;
+            result.fixableWarningCount += 1;
+          }
+        }
+      }
```

Comment:
Could we add some breathing room in the form of blank lines here? It helps for readability.

Reply:
Done. I also changed the code to return early so nesting is reduced to improve readability.